### PR TITLE
🎨 Dark mode when user prefers dark mode

### PR DESCRIPTION
**Pull request by alekangelov**
Tired of getting blasted with light at night tbh

cheers

added a button that inverts the body

image

image

### CODE REVIEW

**Comment by stappersg** [comment]
Acknowledge on the wish of _"dark mode"_.

The first attempt to implement it, should be discarded.

File: backend/src/main.rs
Comment (stappersg) [inline backend/src/main.rs:1]: These removals of empty lines are **NOT related** to `dark mode` change.

File: frontend/style.scss
Comment (stappersg) [inline frontend/style.scss:1]: The replacing of `'` with `"` pollutes the dark mode commit.

**Comment by stappersg** [comment]
On Sun, Sep 10, 2023 at 01:24:50AM -0700, Geert Stappers wrote:
> @stappersg commented on this pull request.
> 
> On frontend/style.scss:
> 
> The replacing of `'` with `"` pollutes the dark mode commit. 
> 

Replacing "empty lines that are separators to improve readability"
with new CSS lines should be just adding new CSS lines.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by stappersg** [comment]
odd icon

Use another icon as _trash can_ for the `invert body` button.

File: frontend/style.scss
Comment (alekangelov) [reply frontend/style.scss:1]: my bad too, I have a default formatter across the board should be fine now

File: backend/src/main.rs
Comment (alekangelov) [reply backend/src/main.rs:1]: my bad, had some logs there 👍

**Comment by alekangelov** [comment]
> Acknowledge on the wish of _"dark mode"_.
> 
> The first attempt to implement it, should be discarded.

so just discard it and reimplement?

**Comment by alekangelov** [comment]
> > Acknowledge on the wish of _"dark mode"_.
> > The first attempt to implement it, should be discarded.
> 
> so just discard it and reimplement?

I just removed the icon all together, I can put in a little sun or something

**Comment by stappersg** [comment]
On Sun, Sep 10, 2023 at 01:52:50AM -0700, Alek Angelov wrote:
> On Sun, Sep 10, 2023 at 01:12:50AM -0700, Geert Stappers wrote:
> > 
> > The first attempt to implement it, should be discarded.
> 
> so just discard it and reimplement?
> 

Too late

**Comment by stappersg** [comment]
> I can put in a little sun or something

_just do it_

**Comment by stappersg** [comment]
' versus " awareness

Make another commit to avoid pollution of this merge request.

https://github.com/tweedegolf/mailcrab/pull/102/files currently shows

' replaced by "

**Comment by alekangelov** [comment]
squashed the commits, should be 1 commit now with ' instead of " and all the changes requested

**Comment by stappersg** [comment]
On Sun, Sep 10, 2023 at 06:30:50AM -0700, Alek Angelov wrote:
> squashed the commits, should be 1 commit

Acknowledge.

What about a commit message?


> now with ' instead of " 

Ack


> and all the changes requested


No, that is (currently) not true.



Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by stappersg** [comment]
On Sun, Sep 10, 2023 at 07:02:06AM -0700, Geert Stappers wrote:
> On Sun, Sep 10, 2023 at 06:30:50AM -0700, Alek Angelov wrote:
> > squashed the commits, should be 1 commit
> 
> Acknowledge.
> 
> What about a commit message?
> 
> 
> > and all the changes requested
> 
> 
> No, that is (currently) not true.
> 
> 

@marlonbaeten  please postpone merging this feature.

For various reasons:
* #103 should go first (and this MR rebased)
* there are pending changes
* lack of a commit message


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by alekangelov** [comment]
why in god's name would you offend me this makes absolutely no sense, im literally just trying to help out

**Comment by alekangelov** [comment]
the hostility here is insane

**Comment by alekangelov** [comment]
image

f this, i wont be called **_that_** for literally contributing to something

**Comment by marlonbaeten** [comment]
@alekangelov thank you for this! I will merge your work this morning and release it in version 0.3 Sorry for being late to respond to your PR - your work is very much appreciated!

@stappersg please be more welcoming to any contributions, and more friendly to their contributors. I think your trying to help to keep the commit history clean, but in the end it's for me to decide what to merge. I value an open and friendly environment for any contributions way more than a clean commit history.

**Comment by marlonbaeten** [comment]
Merged together with #104

**Comment by stappersg** [comment]
> why in god's name would you offend me this makes absolutely no sense

Rethink that.



> im literally just trying to help out

My point of view:  I did see a fresh idea (switching light mode dark mode) with lots of fresh paint, rough edges and other signs of a hasty release.


> the hostility here is insane

Illustrated by the re-opening of a revoked merge request.
